weld/vectorizer.rs

```diff
@@ -232,40 +245,91 @@ pub fn get_id_element(sk: &ScalarKind, op: &BinOpKind) -> WeldResult<Option<Expr
             return Ok(None);
         }
     };
-    
+
     Ok(Some(identity))
 }
 
+fn make_select_for_kv(cond:  Expr<Type>,
+                      kv:    Expr<Type>,
+                      ident: Expr<Type>) -> WeldResult<Option<Expr<Type>>> {
+    if let MakeStruct{ ref elems } = kv.kind {
+        let mut sym_gen = SymbolGenerator::from_expression(&kv);
+        let name = sym_gen.new_symbol("k");
+        let kv_true = exprs::makestruct_expr(vec![elems[0].clone(), elems[1].clone()])?;
```

Comment:
This needs a slight modification: change it to:

```rust
 let kv_struct = exprs::ident_expr(name, kv.ty.clone())?;
 let kv_true = kv_struct.clone(); // use the original
 let kv_ident = exprs::makestruct_expr(vec![exprs::getfield_expr(kv_struct, 0)?, ident])?; // use the original key and the identity as the value
```

Right now the key is still being copied since the name isn't actually being used - same with value, since the `Let` expression evaluates `kv`, and then `elems[0]` and `elems[1]` are being pulled out and re-evaluted.